src/consensus/raft/raft.h

```diff
@@ -20,28 +20,50 @@ namespace raft
 {
   class TermHistory
   {
+    // Entry i stores the first index in term i+1
```

Comment:
The old behaviour was `Entry i stores the first index in term i`. Since term 0 doesn't really exist, we always had a duplicate first entry (`[2, 2, 5, 10, ...]`). This removes that unused first entry, and shifts the off-by-ones in the code.